### PR TITLE
Adding readme to the mock package

### PR DESCRIPTION
**Pull request by igor-ribeiiro**
Added readme and ran the `lint:fix` to the `office-addin-mock` package.

### CODE REVIEW

**Comment by millerds** [comment]
Did you get the text for this from a content writer and/or PM?  We really should have them providing the readme content.

**Comment by igor-ribeiiro** [comment]
> Did you get the text for this from a content writer and/or PM? We really should have them providing the readme content.

I created it. I just thought we needed to have something before sending for people to try it out. 

@Rick-Kirkham could you (or someone from the content team), review the content of this readme? 
It is related to the new mocking package we are developing to help unit test with Office-js.

File: packages/office-addin-mock/README.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+```Typescript
+import { OfficeMockObject } from "office-addin-mock";
+
+async function run() {
+  try {
+    await Excel.run(async (context) => {
+      /**
+       * Insert your Excel code here
+       */
+      const range = context.workbook.getSelectedRange();
```
Comment (Rick-Kirkham) [inline packages/office-addin-mock/README.md:88]: If we're serious about TypeScript, should this be:

```suggestion
      const range: Excel.Range = context.workbook.getSelectedRange();
```

File: packages/office-addin-mock/README.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        },
+      },
+      getSelectedRange: function () {
+        return this.range;
+      },
+    },
+  },
+};
+
+describe(`Run`, function () {
```
Comment (Rick-Kirkham) [inline packages/office-addin-mock/README.md:120]: Does it matter that "Run" is capitalized here, but lowercase where the function is declared?

File: packages/office-addin-mock/README.md [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+  it("Using json", async function () {
+    const excelMock = new OfficeMockObject(MockData) as any;
+    excelMock.addMockFunction("run", async function (callback) {
+      await callback(excelMock.context);
+    });
+    global.Excel = excelMock;
+    await run();
+    assert.strictEqual(excelMock.context.workbook.range.format.fill.color, "yellow");
```
Comment (Rick-Kirkham) [inline packages/office-addin-mock/README.md:128]: So, as I understand it, the test is only verifying that the string "yellow" was written to a proxy object, it doesn't test that a cell in an Excel document was turned yellow. I think we should add a brief paragraph to the top of Example 2 that reminds the reader of the peculiar "proxy" nature of Office objects and that the example is not testing actual interaction with a workbook.

**Review by Rick-Kirkham — CHANGES_REQUESTED** [review]

File: packages/office-addin-mock/README.md [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+- Office-js API needs to be loaded by an HTML file, making the API not available beforehand.
+- Some test APIs may require the entire object to be mocked, which can have more than 100 properties, making it not feasible.
+- Tests need to preserve the order of the functions `load` or `sync`, which are difficult to test because stateless test APIs do not support easily adding state variables to handle those functions.
+
+## Usage
+
+Import `office-addin-mock` to your testing file:
+
+```Typescript
```
[the remark this answers is not in the document]
Comment (igor-ribeiiro) [reply packages/office-addin-mock/README.md:15]: Nice catch, I will talk to him to see what our approach is. I guess we should have the default to be Javascript, with some Typescript examples only in the Official docs, I will see what he says about it.

File: packages/office-addin-mock/README.md [hunk 5]
```diff
@@ -0,0 +1,131 @@
+# Office-Addin-Mock
+
+  This package provides an easier way to unit test the Office-js API.
+  
+  It aims to solve problems that arise when trying to mock the API`s objects:
+
+- Office-js API needs to be loaded by an HTML file, making the API not available beforehand.
+- Some test APIs may require the entire object to be mocked, which can have more than 100 properties, making it not feasible.
```
[the remark this answers is not in the document]
Comment (igor-ribeiiro) [reply packages/office-addin-mock/README.md:8]: Should be _mocking_. Just updated it

File: packages/office-addin-mock/README.md [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+- Tests need to preserve the order of the functions `load` or `sync`, which are difficult to test because stateless test APIs do not support easily adding state variables to handle those functions.
+
```
[the remark this answers is not in the document]
Comment (igor-ribeiiro) [reply packages/office-addin-mock/README.md:10]: You are right, I added an `Installation section to the README.

File: packages/office-addin-mock/README.md [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+    range: {
+      address: "C2",
+    },
+    getSelectedRange: function () {
+      return this.range;
+    },
+  },
+};
+
+describe(`getSelectedRangeAddress`, function () {
```
[the remark this answers is not in the document]
Comment (igor-ribeiiro) [reply packages/office-addin-mock/README.md:69]: Describe comes from Mocha and is not really necessary to use with this package, you could use any other testing framework as well, like Jest for example.
Do you think it is fine for me to just put the installation to `mocha` (or a link to their documentation), or just a disclaimer saying use whatever you want?

File: packages/office-addin-mock/README.md [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+
+  This package provides an easier way to unit test the Office-js API.
+  
+  It aims to solve problems that arise when trying to mock the API`s objects:
+
+- Office-js API needs to be loaded by an HTML file, making the API not available beforehand.
+- Some test APIs may require the entire object to be mocked, which can have more than 100 properties, making it not feasible.
+- Tests need to preserve the order of the functions `load` or `sync`, which are difficult to test because stateless test APIs do not support easily adding state variables to handle those functions.
+
+## Usage
```
[the remark this answers is not in the document]
Comment (igor-ribeiiro) [reply packages/office-addin-mock/README.md:11]: Is "indented note" some syntax specific to markdown (I think I'm not aware)?
I added a small text to mention that we are using Mocha as the testing framework.

File: packages/office-addin-mock/README.md (hunk 2 above)
Comment (igor-ribeiiro) [reply packages/office-addin-mock/README.md:120]: It doesn't. The "Run" here is just a text that will get outputted when the tests are run. That's what the developers will see:
image

File: packages/office-addin-mock/README.md (hunk 3 above)
Comment (igor-ribeiiro) [reply packages/office-addin-mock/README.md:128]: You are right, we have no interaction with Excel here. Actually, this whole package won't have any interaction with Office as it is just for unit testing.
I added this disclaimer right at the beginning of the file.

**Comment by igor-ribeiiro** [comment]
@Rick-Kirkham the PR is ready to be reviewed again :D

File: packages/office-addin-mock/README.md [hunk 9]
```diff
@@ -0,0 +1,229 @@
+# Office-Addin-Mock
+
+  This library provides an easier way to unit test the Office JavaScript library (hereafter, "Office-js") API.
+  This library does not depend on Office, so it doesn't tests actual interactions with it.
```
Comment (Rick-Kirkham) [inline packages/office-addin-mock/README.md:4]:
```suggestion
  This library does not depend on Office, so it doesn't test actual interactions with it.
```

**Review by Rick-Kirkham — DISMISSED** [review]
One little typo. But I'm approving.

**Pull request merged**